frontend/public/components/operator-hub/operator-hub-subscribe.tsx

```diff
@@ -51,12 +51,12 @@ export const OperatorHubSubscribeForm = withFormState((props: OperatorHubSubscri
   }
 
   const {provider, channels = [], packageName} = props.packageManifest.data.status;
-  const srcProvider = _.get(props.packageManifest.data, 'metadata.labels.opsrc-provider');
+  const srcProvider = _.get(props.packageManifest.data, 'metadata.labels.opsrc-provider', 'custom');
   const providerType = getOperatorProviderType(props.packageManifest.data);
 
   const submit = () => {
     const operatorGroupNamespace = props.operatorGroup.data.find(og => og.metadata.name === props.formState().target).metadata.namespace;
-    const OPERATOR_HUB_CSC_NAME = `${OPERATOR_HUB_CSC_BASE}-${srcProvider}`;
+    const OPERATOR_HUB_CSC_NAME = `${OPERATOR_HUB_CSC_BASE}-${srcProvider}-${props.operatorGroup}`;
```

Comment:
Is this guaranteed to be unique? (I don't know enough about Operator Hub to say.)

Note you could use `generateName: foo-` instead of `name: foo` which will append a few random characters to the end.

Reply:
Yeah I thought of this but I need to query the name when I update the resource to add a package. The problem with the implementation I had before was that if you tried to subscribe to an operator of the same provider type but to a different operator group, the targetNamespace field of the CSC would be updated (instead of just the packages section). This would cascade delete the catalogsource and configmaps that had been created previously with this CSC. Hope this helps clarify the issue a bit better!